{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c8579a33-bc5f-4c4a-9a4d-1b03d253c897",
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "import re\n",
    "import numpy as np\n",
    "import random\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib as mpl\n",
    "import matplotlib.patches as mpatches\n",
    "mpl.rcParams['pdf.fonttype'] = 42\n",
    "mpl.rcParams['font.size'] = 12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fb8937d6-6553-458d-8e1f-7ed4a5f18d0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "dt_outpath = '../data/'\n",
    "fig_outpath = '../figures/'\n",
    "bash_outpath = './sh/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e7dbb12-1729-4549-8d90-4e017c36c11b",
   "metadata": {},
   "source": [
    "## STAR 2 step alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4ee6dfa5-6a5a-4b10-86ba-98a8624c8d7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_ref = '/private/groups/brookslab/gabai/tools/ref/yst/star_aresv13/'\n",
    "path_to_fastq = '/private/groups/brookslab/data.rep/yeast_shortread_rnaseq/'\n",
    "outpath = '/private/groups/brookslab/gabai/projects/yeastMeth/data/rna/totalrna/rsem/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "04000ca0-c34b-4657-aef8-afe0dc84c06f",
   "metadata": {},
   "outputs": [],
   "source": [
    "fs = os.listdir(path_to_fastq)\n",
    "fastq_prefix = [re.split(r'_R.*_001\\.fastq\\.gz', f)[0] for f in fs if f.endswith('fastq.gz')]\n",
    "fastq_prefix = sorted(set(fastq_prefix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c903b0cb-e797-4f5b-9b10-0e5198eec696",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = ['ys18_rep1', 'ys18_rep2', 'ys18_rep3', 'ym209_rep1', 'ym209_rep2', 'ym209_rep3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ba074ecd-5110-45b3-beca-0944f4ac9638",
   "metadata": {},
   "outputs": [],
   "source": [
    "for s, fq in zip(samples, fastq_prefix):\n",
    "    outfile= open(f'{bash_outpath}250804_runSTAR_2step_on_aresv13_{s}.sh', 'w')\n",
    "    header = f'#!/bin/bash\\n#SBATCH --job-name=yeast_totalRNA_STAR_step1_%s\\n#SBATCH --ntasks=1\\n#SBATCH --cpus-per-task=32\\n#SBATCH --nodes=1\\n#SBATCH --time=120\\n#SBATCH --mem=50G\\n#SBATCH --partition=medium\\n#SBATCH --error={outpath}log/lsf_%s.err\\n#SBATCH --output={outpath}log/lsf_%s.out\\n#SBATCH --mail-type=END,FAIL\\n#SBATCH --mail-user=gabai@ucsc.edu\\n' % (s, s, s)\n",
    "    cmd1 = f'/private/groups/brookslab/bin/STAR-2.7.11b/bin/Linux_x86_64_static/STAR --runThreadN 16 --genomeDir {path_to_ref} --readFilesIn {path_to_fastq}{fq}_R1_001.fastq.gz {path_to_fastq}{fq}_R2_001.fastq.gz {precise_params} --outFileNamePrefix {outpath}{s}\\n'\n",
    "    outfile.write(header)\n",
    "    outfile.write('\\n')\n",
    "    outfile.write(cmd1)\n",
    "    outfile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4505046b-b0ed-40c8-b317-37f8ac9bc9f8",
   "metadata": {},
   "source": [
    "## RSEM quantification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "759a5624-2461-417a-a8cc-be8067c0f4f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_bam = '/private/groups/brookslab/gabai/projects/yeastMeth/data/rna/totalrna/star_2step/'\n",
    "path_to_rsem_ref = '/private/groups/brookslab/gabai/tools/ref/yst/rsem_ares_v13'\n",
    "outpath = '/private/groups/brookslab/gabai/projects/yeastMeth/data/rna/totalrna/rsem/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b2382379-d64c-4299-97e0-9e7905396db4",
   "metadata": {},
   "outputs": [],
   "source": [
    "precise_params = f\"\"\"--no-bam-output \\\\\n",
    "    --seed 12345 -p 16 \\\\\n",
    "    --strandedness reverse \\\\\n",
    "    --alignments \\\\\n",
    "    --paired-end\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d3679e52-d9f3-4430-97d0-195baa671cca",
   "metadata": {},
   "outputs": [],
   "source": [
    "for s in samples:\n",
    "    outfile= open(bash_outpath + '250807_totalrna_rsem_quant_%s.sh'%s, 'w')\n",
    "    header = f'#!/bin/bash\\n#SBATCH --job-name=yeast_totalrna_rsem_%s\\n#SBATCH --ntasks=1\\n#SBATCH --cpus-per-task=20\\n#SBATCH --nodes=1\\n#SBATCH --time=120\\n#SBATCH --mem=100G\\n#SBATCH --partition=medium\\n#SBATCH --error={outpath}log/lsf_%s.err\\n#SBATCH --output={outpath}log/lsf_%s.out\\n#SBATCH --mail-type=END,FAIL\\n#SBATCH --mail-user=gabai@ucsc.edu\\n' % (s, s, s)\n",
    "    cmd = f'/private/groups/brookslab/bin/RSEM-1.3.0/rsem-calculate-expression {precise_params} {path_to_bam}{s}Aligned.toTranscriptome.out.pass.sorted.pass.rname.sorted.bam {path_to_rsem_ref} {outpath}{s}'\n",
    "    outfile.write(header)\n",
    "    outfile.write('\\n')\n",
    "    outfile.write(cmd)\n",
    "    outfile.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
